internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -113,19 +116,38 @@ spec:
         for PULL_SPEC in ${PULL_SPEC_LIST}; do
             # make sure they query is done using the internal name instead of the public
             PULL_SPEC=$(sed  's/registry.redhat.io/registry.access.redhat.com/g' <<< $PULL_SPEC)
-            FBCOptIn+=(`isFBCOptIn ${PULL_SPEC}`)
+
+            echo "Attempting to fetch from ${FETCH_URL} to check if fragment is \`fbc_opt_in==true\`..."
+            fbcOptIn+=(`isfbcOptIn ${PULL_SPEC}`)
         done
-        FBCOptIn=$(echo ${FBCOptIn} |sort |uniq)
 
-        # in case of more than one image all should have the same fbc_opt_in value
-        if [ $(wc -w <<< ${FBCOptIn[@]}) -eq 1 ]; then
-            echo -en "fbc_opt_in=${FBCOptIn}" | tee $(results.genericResult.path)
+        fbcOptIn=$(echo ${fbcOptIn} |sort |uniq)
+        mustPublishIndexImage=$(echo ${fbcOptIn} |sort |uniq)
+        mustSignIndexImage=$(echo ${fbcOptIn} |sort |uniq)
+
+        if [ "$(params.hotfix)" == "true" ]; then
+            echo "Hotfix build"
+            fbcOptIn="false"
+            mustSignIndexImage="true"
+            mustPublishIndexImage="true"
         else
-            FBCOptIn="false"
-            echo -en "fbc_opt_in=false" | tee $(results.genericResult.path)
+            echo "Fragment has \`fbc_opt_in==${fbcOptIn}\`"
+            # in case of more than one image all should have the same fbc_opt_in value
+            if [ $(wc -w <<< ${fbcOptIn[@]}) -ne 1 ]; then
+                fbcOptIn="false"
+                mustSignIndexImage="true"
+                mustPublishIndexImage="true"
+            fi
         fi
-
-        echo "Fragment has \`fbc_opt_in==${FBCOptIn}\`"
+        jq --arg fbc_opt_in "${fbcOptIn}" \
+           --arg publish_index_image "${mustPublishIndexImage}" \
+           --arg sign_index_image "${mustSignIndexImage}" \
+           '{
+             "fbc_opt_in": $fbc_opt_in,
+             "publish_index_image": $publish_index_image,
+             "sign_index_image": $sign_index_image
+            } | tostring' <<< "{}" \
```

Comment:
About the escapes. There was some case in the past that passing it as json, openshift was reading the field as object instead of text. For instance, this did not happen in my local env, but when the version went to stage, it broke there. As I am not sure this is still happening (due to different Openshift Pipelines version and Openshift version I guess)  I preferred to keep it this way just in case.

Reply:
About the condensed version, I might add the -c to transfer the result, but the formatting it was more because the _one line_ version passes the 120 cols.

Reply:
About the escaping, it's weird that it would be treated as something other than a string. But ok, if that's the case, then it makes sense.

OK, it doesn't really matter if it's on one line or several lines. But you can definitely use `-n` instead of `<<< {}`.

Reply:
Unf. there was no ticket for it. If you want I can share the talk I had with @scoheb about it.

Reply:
Maybe the solution used was not the best, but the output was the one desired. 

```
$ jq -n -cr '{ "state": .state, "state_reason": .state_reason }'
{"state":null,"state_reason":null}
$ jq -n -cr '{ "state": .state, "state_reason": .state_reason }'|jq -Rc
"{\"state\":null,\"state_reason\":null}"
```

Reply:
Ah, I see. So the first `jq` composes some json and prints it. And the second one treats it as plain string and prints it back. Quite cumbersome if you ask me :) Just adding `|tostring` to the first `jq` would probably have the same effect, right?

Anyway, hopefully this is not needed anymore.

Reply:
Indeed. It was literally a quick fix without much review just move forward with the FBC project at the time. This task actually needs refactory and tests (the entire repo).

Reply:
OK, cool.

Reply:
I removed the ` <<<` in favor of the `-n` but I still kept the ` tostring` for the moment.

Reply:
I don't see any `tostring` anymore.

Reply:
My bad. I forgot to commit and push.